File: app/src/main/java/eu/kanade/tachiyomi/data/database/tables/MangaCategoryTable.java
package eu.kanade.tachiyomi.data.database.tables;

import android.support.annotation.NonNull;

public class MangaCategoryTable {

    @NonNull
    public static final String TABLE = "mangas_categories";

    @NonNull
    public static final String COLUMN_ID = "_id";

    @NonNull
    public static final String COLUMN_MANGA_ID = "manga_id";

    @NonNull
    public static final String COLUMN_CATEGORY_ID = "category_id";

    // This is just class with Meta Data, we don't need instances
    private MangaCategoryTable() {
        throw new IllegalStateException("No instances please");
    }

    // Better than static final field -> allows VM to unload useless String
    // Because you need this string only once per application life on the device
    @NonNull
    public static String getCreateTableQuery() {
        return "CREATE TABLE " + TABLE + "("
                + COLUMN_ID + " INTEGER NOT NULL PRIMARY KEY, "
                + COLUMN_MANGA_ID + " INTEGER NOT NULL, "
                + COLUMN_CATEGORY_ID + " INTEGER NOT NULL, "
                + "FOREIGN KEY(" + COLUMN_CATEGORY_ID + ") REFERENCES " + CategoryTable.TABLE + "(" + CategoryTable.COLUMN_ID + ") "
                + "ON DELETE CASCADE, "
                + "FOREIGN KEY(" + COLUMN_MANGA_ID + ") REFERENCES " + MangaTable.TABLE + "(" + MangaTable.COLUMN_ID + ") "
                + "ON DELETE CASCADE"
                + ");";

    }

}

File: app/src/main/java/eu/kanade/tachiyomi/data/database/tables/MangaCategoryTable.kt
package eu.kanade.tachiyomi.data.database.tables

object MangaCategoryTable {

    const val TABLE = "mangas_categories"

    const val COL_ID = "_id"

    const val COL_MANGA_ID = "manga_id"

    const val COL_CATEGORY_ID = "category_id"

    val createTableQuery: String
        get() =
            """CREATE TABLE $TABLE(
            $COL_ID INTEGER NOT NULL PRIMARY KEY,
            $COL_MANGA_ID INTEGER NOT NULL,
            $COL_CATEGORY_ID INTEGER NOT NULL,
            FOREIGN KEY($COL_CATEGORY_ID) REFERENCES ${CategoryTable.TABLE} (${CategoryTable.COL_ID})
            ON DELETE CASCADE,
            FOREIGN KEY($COL_MANGA_ID) REFERENCES ${MangaTable.TABLE} (${MangaTable.COL_ID})
            ON DELETE CASCADE
            )"""
}
